setup/providers/appengine.md

```diff
@@ -1,37 +1,76 @@
-    * Go to the API Management page.
-    * Enable the **App Engine Admin** and **Compute Engine APIs**.
+```bash
+gcloud info
+```
+
+If this is your first time deploying to App Engine in your project, create an App Engine application. 
+You cannot change your application's region, so pick wisely:
+
+```bash
+gcloud app create --region <e.g., us-central>
+```
+
+You will also need to enable the App Engine Admin API for your project:
+
+```bash
+gcloud service-management enable appengine.googleapis.com
+```
 
-2. If this is your first time deploying to App Engine in your project, create an App Engine application using 
-   `gcloud`. You cannot change your application's region, so pick wisely:
+## Downloading Credentials
 
-   ```bash
-   gcloud app create --region <e.g., us-central>
-   ```
-3. Spinnaker does not need to be given service account credentials if it is running on a Google Compute Engine VM _and_
-   it is deploying to an App Engine application inside the same Google Cloud Platform project in which it is running. If Spinnaker
-   will need service account credentials, follow these steps for the project you would like to deploy to:
-    * Inside the [Google Cloud Console](https://console.cloud.google.com), go to the Credentials tab
-     on the API Management page.
-    * Select the **Service account key** item from the **Create credentials** menu.
-    * Select a service account, the **JSON** key type, and click **Create**.
-    * Safeguard the JSON file that your browser will download.
+Spinnaker does not need to be given [service account](https://cloud.google.com/compute/docs/access/service-accounts)
+credentials if it is running on a Google Compute Engine VM whose
+application default credentials have sufficient scopes to deploy to App Engine _and_ 
+Spinnaker is deploying to an App Engine application inside the same Google Cloud Platform project in which it is running. If
+Spinnaker will not need to be given service account credentials, or if you already have such a service account 
+with the corresponding JSON key downloaded, skip ahead to [Adding an Account](#adding-an-account).
+
+Run the following commands to create a service account 
+with the `roles/appengine.appAdmin` and `roles/storage.admin` roles enabled:
+
+```bash
+SERVICE_ACCOUNT_NAME=spinnaker-appengine-account
+SERVICE_ACCOUNT_DEST=~/.gcp/appengine-account.json
+
+gcloud iam service-accounts create \
+    $SERVICE_ACCOUNT_NAME \
+    --display-name $SERVICE_ACCOUNT_NAME
+
+SA_EMAIL=$(gcloud iam service-accounts list \
+    --filter="displayName:$SERVICE_ACCOUNT_NAME" \
+    --format='value(email)')
+
+PROJECT=$(gcloud info --format='value(config.project)')
+
+gcloud projects add-iam-policy-binding $PROJECT \
+    --role roles/appengine.appAdmin \
+    --role roles/storage.admin \
+    --member serviceAccount:$SA_EMAIL
+
+mkdir -p $(dirname $SERVICE_ACCOUNT_DEST)
+
+gcloud iam service-accounts keys create $SERVICE_ACCOUNT_DEST \
+    --iam-account $SA_EMAIL
+```
 
```

Comment:
I would leave a comment here making it explicit that your key now sits in `$SERVICE_ACCOUNT_DEST`